new-packages/ts-project/src/state.ts

```diff
@@ -339,329 +322,323 @@ export function extractState(
-        continue;
+  forEachStaticProperty(ctx, ts, state, (prop, key) => {
+    switch (key) {
+      case 'id': {
+        if (ts.isStringLiteralLike(prop.initializer)) {
+          ctx.idMap[prop.initializer.text] = node.uniqueId;
+          return;
+        }
+        return;
       }
-      seen.add(key);
-
-      switch (key) {
-        case 'id': {
-          if (ts.isStringLiteralLike(prop.initializer)) {
-            ctx.idMap[prop.initializer.text] = node.uniqueId;
-            continue;
-          }
-          break;
+      case 'context': {
+        if (parentId !== undefined) {
+          ctx.errors.push({
+            type: 'state_property_invalid',
+          });
+          return;
         }
-        case 'context': {
-          if (parentId !== undefined) {
-            ctx.errors.push({
-              type: 'state_property_invalid',
-            });
-            continue;
-          }
-          if (ts.isObjectLiteralExpression(prop.initializer)) {
-            const obj = getJsonObject(ctx, ts, prop.initializer);
-            if (obj) {
-              ctx.digraph.data.context = obj;
-            }
-            continue;
-          }
-          if (
-            ts.isFunctionExpression(prop.initializer) ||
-            ts.isArrowFunction(prop.initializer)
-          ) {
-            ctx.digraph.data.context = `{{${prop.initializer.getText(
-              ctx.sourceFile,
-            )}}}`;
-            continue;
+        if (ts.isObjectLiteralExpression(prop.initializer)) {
+          const obj = getJsonObject(ctx, ts, prop.initializer);
+          if (obj) {
+            ctx.digraph.data.context = obj;
           }
+          return;
+        }
+        if (
+          ts.isFunctionExpression(prop.initializer) ||
+          ts.isArrowFunction(prop.initializer)
+        ) {
+          ctx.digraph.data.context = `{{${prop.initializer.getText(
+            ctx.sourceFile,
+          )}}}`;
+          return;
+        }
 
+        ctx.errors.push({ type: 'state_property_unhandled' });
+        return;
+      }
+      case 'always': {
+        extractEdgeGroup(ctx, ts, prop, {
+          sourceId: node.uniqueId,
+          eventTypeData: { type: 'always' },
+        });
+        return;
+      }
+      case 'on': {
+        if (!ts.isObjectLiteralExpression(prop.initializer)) {
           ctx.errors.push({ type: 'state_property_unhandled' });
-          break;
+          return;
         }
-        case 'always': {
-          if (!ts.isPropertyAssignment(prop)) {
-            ctx.errors.push({ type: 'state_property_unhandled' });
+        for (const transition of prop.initializer.properties) {
+          if (!ts.isPropertyAssignment(transition)) {
+            ctx.errors.push({ type: 'transition_property_unhandled' });
+            return;
+          }
+          const event = getPropertyKey(ctx, ts, transition);
+          if (!event) {
+            ctx.errors.push({ type: 'transition_property_unhandled' });
```

Comment:
nit: maybe we wan use `forEachStaticPropery` here too.

Reply:
done